test/unit/test_sap_rfc_user.py

```diff
@@ -338,7 +338,52 @@ def test_change_user_ok(self):
 
         retval = self.manager.change_user(self.connection, user_builder)
 
-        self.connection.call.assert_called_once_with('BAPI_USER_CHANGE', **user_builder.build_change_rfc_params())
+        self.assertEqual(self.connection.call.call_count, 2)
+
+        self.connection.call.assert_has_calls([call('BAPI_USER_GET_DETAIL', USERNAME=user_builder.get_username()),
+                                               call('BAPI_USER_CHANGE', **user_builder.build_change_rfc_params())])
+
+        self.assertEqual(str(retval), str(BAPIReturn(self.bapirettab)))
+
+    def test_create_user_and_set_productive_pass_ok(self):
+        user_builder = self.manager.user_builder()
+        user_builder.set_username(self.username)
+        user_builder.set_productive_password(True)
+        user_builder.set_password('UnitTestPass')
```

Comment:
I was also wondering why we should use the method "set_productive_password" instead of an optional parameter of the method "set_password". Could you please explain your decision?

Reply:
I wanted to keep it uniform. We handle all the parameters with a separate  set method. 
However, I have changed it to optional parameter.